osu.Game/Overlays/Chat/ChannelList/ChannelList.cs

```diff
@@ -26,13 +26,16 @@ public class ChannelList : Container
         public Action<Channel>? OnRequestSelect;
         public Action<Channel>? OnRequestLeave;
 
-        public IEnumerable<Channel> Channels => publicChannelFlow.Channels.Concat(privateChannelFlow.Channels);
+        public IEnumerable<Channel> Channels =>
+            announceChannelFlow.Channels.Concat(publicChannelFlow.Channels).Concat(privateChannelFlow.Channels);
```

Comment:
This, as well as the way the announce section is added is kind of ad hoc, and it's a byproduct of how things are structured here. The current structure of the channel list is as follows:

```
ChannelList
    FillFlowContainer
        ChannelListLabel
        ChannelListItemFlow
        ChannelListLabel
        ChannelListItemFlow
        ChannelListLabel
        ChannelListItemFlow
```

which is awkward to work with and less structured than I'd like to see. Add the (very real) possibility of more channel types being supported there and it stops scaling real fast.

I'd like to see some more structuredness here, by doing something like:

```
ChannelList
    FillFlowContainer<ChannelGroup>
        ChannelGroup
             Label
             ChannelListItemFlow
        ChannelGroup
             Label
             ChannelListItemFlow
        ChannelGroup
             Label
             ChannelListItemFlow
```

This solves several concerns:

* Rather than `.Concat()`ting three flows above, you can have `ChannelGroup` expose the inner flow and do something along the lines of `channelGroupContainer.Children.SelectMany(group => group.ChannelFlow.Children)` to get all channels
* `updateVisibility()` reduces to toggling the entire group's alpha rather than the heading separately and the flow separately

Reply:
Makes sense, will update :+1: